src/java/org/apache/cassandra/index/sai/disk/io/FilterIndexInput.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.io;
+
+import java.io.IOException;
+
+import org.apache.lucene.store.IndexInput;
+
+public abstract class FilterIndexInput extends IndexInput
```

Comment:
nit: If this is truly only used in tests, we might just move it to the test folder?

naming nit: Could also be `ForwardingIndexInput`, if you like the way Guava names this stuff

Reply:
I see no reason for this class at all. It's only used by `TrackingIndexInput` so we could just put these methods in there and be done with it. We can probably give that class a better name as well.